src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -41,28 +44,44 @@ class CallActor extends Actor {
 
   private def creator: ActorRef = _creator.get
 
-  override def receive: Receive =
+  private def notStarted: Receive = {
     LoggingReceive {
-      case Start(call, symbolStore) =>
+      case Start(call, backend, symbolStore) =>
         val inputs = symbolStore.locallyQualifiedInputs(call)
         val tryCommandLine = call.task.command.instantiate(inputs)
         tryCommandLine match {
           case Success(commandLine) =>
             _creator = Option(sender())
-            self ! Run(commandLine)
+            self ! Run(commandLine, backend, call, symbolStore)
             creator ! Started
-          case Failure(t) =>
-            sender ! Failed(t)
+            context.become(started)
+          case failure =>
+            sender ! failure
         }
 
-      case bad @_ if !isStarted =>
+      case bad@_ if !isStarted =>
         sender ! InvalidOperation(s"Received message $bad, but not yet Started.")
+    }
+  }
 
-      case Run(commandLine) =>
-        log.info(commandLine)
-        // TODO get actual outputs and send them along in the reply
-        creator ! Done(Map.empty[String, WdlValue])
-        self ! Stop
+  private def started: Receive = {
+    LoggingReceive {
+      case Run(commandLine, backend, call, symbolStore) =>
+        log.info(self + " executing command: " + commandLine)
+        val tryOutputs = backend.executeCommand(commandLine, call, call.task.outputs, symbolStore)
+        val failures = tryOutputs.filter { _._2.isFailure }
```

Comment:
What about something like (warning, hand typed code ahead! Pseudocodeish):

    val failures = tryOutputs.collect {case x: Failure[X] => x.exception.getMessage}
    if (failures.isEmpty) tryOutputs foreach {x => doSomethingWith(x)}
    else doSomethingElseWith(failures)

Note X needs to be the type of the Try

Alternatively, use .partition to get your set of success and failures and do basically what you're doing here